pkg/controller/actions/actions.go

```diff
@@ -1,25 +1,38 @@
 package actions
 
 import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
-func CreateOwnedResource(ctx *core.Context, owner metav1.Object, controlled client.Object) error {
-	_, _, err := createOwnedResource(ctx, owner, controlled)
-	return err
-}
-
 func CreateOrUpdateOwnedResource(ctx *core.Context, owner metav1.Object, controlled client.Object) error {
-	found, gvk, err := createOwnedResource(ctx, owner, controlled)
+
+	err := ctrl.SetControllerReference(owner, controlled, ctx.Scheme)
+	if err != nil {
+		return err
+	}
+
+	gvk, err := getObjectKind(ctx, controlled)
+	if err != nil {
+		return err
+	}
+
+	found := controlled.DeepCopyObject().(client.Object)
```

Comment:
What does this syntax do? `DeepCopyObject().(client.Object)`

Is `found` the right name for this variable? It looks like we perform the actual check on the next line.

Reply:
- The deepCopy method (it's auto-generated) creates a "true" copy of a the data object including proper cloning of all nested/constituent objects. I haven't looked at the details of how this is used here specifically, but that's a normally accepted practice.
- `found` is likely not a right now. It's my copy/paste of Sonny's code (a bad excuse).